ci/setup_env.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+set -eEo pipefail
+
+export IS_A_RELEASE="false"
+export PROD_RELEASE="false"
+export DEV_RELEASE="false"
+export TEST_RELEASE="false"
+export COMMON_FILE_LOCATION='ci/common.sh'
+
+workdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." &> /dev/null && pwd )"
+github_tag="${GITHUB_REF_NAME:-}"
+prod_release_branch="${PROD_RELEASE_BRANCH:-main}"
+prod_release_regex='^[0-9]+\.[0-9]+\.[0-9]+$'
+dev_release_regex='^[0-9]+\.[0-9]+\.[0-9]+(-[0-9]+)?(-rc[0-9]+){1}$'
```

Comment:
Please modify this as:
dev_release_regex='^[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+){1}$'
We want `1.2.3-rc.1` to match, we don't want `1.2.3-rc1` nor `1.2.3-1234-rc.1` to match.

Reply:
From our conversation with @lander86 we agreed on implementing the common nomenclature for releases which is `d.d.d-rc[0-9]*` 

I think we need to meet all together and come up with the solution that everyone agrees on.